### ISSUE TITLE
Environment variables are not loaded from .env files when using "source" from package.json as parcel entry

### ISSUE THREAD

**Issue by jaredtbrown** [parcel-bundler/parcel#7515]
# 🐛 bug report
When using `parcel` or `parcel build` and setting the `source` property in the package.json as the entry file, the `.env` files seem to be ignored as the environment variable values are undefined. However, if the entry is specified in the CLI command, the environment variables resolve correctly.

## 🎛 Configuration (.babelrc, package.json, cli command)
.env
```
TEST_THING=hello
```

package.json
```json
{
 ...
 "source": "src/index.tsx"
 ...
 "scripts": {
   "start": "parcel",
   "build": "parcel build"
 }
 ...
}
```

## 🤔 Expected Behavior

When running the above commands, accessing environment variables should resolve in the value from .env values. IE `process.env.TEST_THING` should return the value of `"hello"`.

## 😯 Current Behavior

When running the above commands, accessing any environment variables resolves with `undefined`.

However, if the the scripts are changed to specify the entry file in the CLI like so:
```json
{
...
"scripts": {
  "start": "parcel src/index.tsx",
  "build": "parcel build src/index.tsx"
}
...
}
```

The environment variables are resolved correctly.

## 🔦 Context

This is not that major of an issue as there is a workaround, but it is unexpected behavior that not everyone may be aware of

## 💻 Code Sample

See above.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.1
| Node             | 16.13.1
| npm/Yarn         | yarn 1.22.10
| Operating System | macOS 12.0.1

**Comment by avrebarra** [comment]
Probably related cmiiw mine also unable to read vars from `.env` for react (js or ts). And still couldnt read it even when using `parcel src/index.html` or using source field in package.json.

`.env` file placed in same level with package.json. No build errors.


Code:
```json
// package.json
{
  "name": "check-parcel-env",
  "version": "1.0.0",
  "license": "MIT",
  "dependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
```

```html
// src/index.html
<!DOCTYPE html>
<html lang="en">
  <body>
    <div id="app"></div>
    <script type="module" src="index.js"></script>
  </body>
</html>
```

```js
// src/index.js
import ReactDOM from "react-dom";

const env = process.env;

if (process.env.NODE_ENV === "development") {
  console.log("Check A:", process.env.NODE_ENV, process.env.API_TOKEN);
  console.log("Check B:", env.NODE_ENV, env.API_TOKEN);
  console.log(env);
}

const app = document.getElementById("app");

ReactDOM.render(<div>app</div>, app);
```

Running`parcel src/index.html`, when opened in browser logs only shows NODE_ENV:
```js
// from chrome's console logs:
Check A: development undefined // ????
Check B: undefined undefined // expected
{} // expected
```

I'm currently using:
- Parcel: 2.0.1
- Node: v17.3.0
- Yarn: 1.22.17
- OS: Ubuntu 20.04.3

**Comment by jbree** [comment]
@avrebarra Parcel performs inlining of env variables at build time. Your code should perform each env access directly from the `process.env` object, rather than creating a new variable `env` like you do in your examples.

See https://parceljs.org/features/node-emulation/#environment-variables

**Comment by avrebarra** [comment]
@jbree yeah I did that in the first line: `console.log("Check A:", process.env.NODE_ENV, process.env.API_TOKEN);`

Like this:
```js
// src/index.js
import ReactDOM from "react-dom";

if (process.env.NODE_ENV === "development") {
  console.log("Check A:", process.env.NODE_ENV, process.env.API_TOKEN);
}

const app = document.getElementById("app");

ReactDOM.render(<div>app</div>, app);
```

when opened in browser logs only shows valid value for NODE_ENV:
```js
// from chrome's console logs:
Check A: development undefined // ????
```

**Comment by jbree** [comment]
Hmm your example works fine for me. I tried it yday on 2.0.1 but again today on 2.1.0. Could it be a syntax error in your .env? Perhaps you could share that one as well for completeness

**Comment by avrebarra** [comment]
@jbree I tried it again on different machine today (my prev notebook somehow got busted) and it's working nicely now. I dont understand, I think I wrote the env correctly. But I guess for my case it's solved for now.

node v17.3.0
yarn 1.22.17
parcel 2.0.1

EDIT: After retrying, it might be because I added the vars but dont rebuild the parcel explicitly (`Ctrl+C` and run `parcel` again). Just realized that vars updates shows up on explicit rebuild, but not when on rebuild triggered by watcher.
